pkg/scalers/cpu_memory_scaler.go

```diff
@@ -50,15 +73,38 @@ func NewCPUMemoryScaler(resourceName v1.ResourceName, config *scalersconfig.Scal
 	}
 
 	return &cpuMemoryScaler{
-		metadata:     meta,
-		resourceName: resourceName,
-		logger:       logger,
+		metadata:      meta,
+		resourceName:  resourceName,
+		logger:        logger,
+		kubeClient:    kubeClient,
+		metricsClient: metricsClient,
 	}, nil
 }
 
-func parseResourceMetadata(config *scalersconfig.ScalerConfig, logger logr.Logger) (cpuMemoryMetadata, error) {
-	meta := cpuMemoryMetadata{}
-	err := config.TypedConfig(&meta)
+func getScaleTarget(scalableObjectName, scalableObjectNamespace string, kubeClient client.Client) (string, string, error) {
+	scaledObject := &kedav1alpha1.ScaledObject{}
+	err := kubeClient.Get(context.Background(), types.NamespacedName{
```

Comment:
is there a specific reason to use `context.Background()`? If not, then we should pass ctx from the top level here.

Reply:
I used `context.Background()` because neither of its top level functions `parseResourceMetadata()` and `NewCPUMemoryScaler` has `context`. If there is any other way, I'd be happy to know

Reply:
feel free to add the param there, you can see that ctx is also added to a few other scalers